File: DSA problems in c++/insertionsort.cpp
 void insert(int arr[], int i)
    {
        // Your code here  
    }
    //Function to sort the array using insertion sort algorithm.
    void insertionSort(int arr[], int n)
    {
        //code here
        for(int i = 1; i < n; i++)      //loop for the rounds and i = 1  because we have to assume that the first element is already sorted
        {
            int temp = arr[i];
            int j = i - 1;
            for(; j >=0; j--)
            {
                if(arr[j] > temp)
                {       //shift occurs
                    arr[j+1] = temp;
                }
                else
                    break;
            }
            arr[j+1] = temp;
        }
    }

File: InsertionSort.java
//Program for Insertion Sort
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        int[] arr = {4,5,2,14,12,34};
        insertionSort(arr);
        System.out.println(Arrays.toString(arr));
    }
    static void insertionSort(int[] arr){
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = i+1; j > 0 ; j--) {
                if(arr[j] < arr[j-1]){
                    swapArray(arr,j,j-1);
                }else {
                    break;
                }
            }
        }
    }
    static void swapArray(int[] arr,int first,int second){
        int temp = arr[first];
        arr[first] = arr[second];
        arr[second] = temp;
    }
}
